modules/Configuration.py

```diff
@@ -0,0 +1,32 @@
+import logging
+import yaml
+
+logger = logging.getLogger('esxi-exporter')
+
+try:
+    logger.debug('loading config.yaml')
+    with open('config.yaml', 'r', encoding='utf8') as f:
+        data = yaml.safe_load(f)
+
+    port: int = int(data['port'])
+    cachetime: int = int(data['cachetime'])
+    blacklisttime: str = int(data['blacklisttime'])
+    ssh_services: list = data['ssh_collector']['services']
+    collectors: list = data['collectors']
+    ssh_threads: int = int(data['ssh_collector']['thread_count'])
+    overallstate_threads: int = int(
+        data['overall_state_collector']['thread_count'])
+    del data
+    logger.debug('config.yaml loaded.')
+
+except KeyError as ex:
+    logger.critical("Missing configuration entry: %s" % str(ex))
+    exit(0)
+
+except IOError as ex:
+    logging.critical("Could not open config.yaml: %s" % str(ex))
+    exit(0)
+
+except TypeError as ex:
+    logging.critical("Could not convert type: %s" % str(ex))
+    exit(0)
```

Comment:
I can see the thought process behind this, but it is very ineffective. Everytime you are calling `config` from your input it basically runs this whole set of code here. I would rather suggest a minimal example. Instantiate a new class `Configuration.py` in your main function and pass this instance to the Collectors. It should have all config options as class attributes.
The collectors then should take the route down the tree with their own name to get their own information. This way it is only read once.